logger.go

```diff
@@ -132,3 +133,74 @@ func formatMessage(msg *irc.Message) string {
 		return ""
 	}
 }
+
+type historyMessage struct {
```

Comment:
Maybe we could just use `irc.Message` here instead?

Reply:
Right. Using historyMessage is a tad more efficient because we skip formatting the time of that message, but even though this is a performance-important part of the code because we call this on a lot of messages, it's probably not worth the effort using that intermediate struct.